dali/base/dasds.cpp

```diff
@@ -289,6 +290,119 @@ class CLCWriteLockBlock : public CLCLockBlock
+                continue;
+            unsigned filterSize = 2;
+            DALockFilterType filterType = (DALockFilterType) atoi(filterTypeStr);
+            switch(filterType)
+            {
+            case DALFTLockType:
+                if (filterFieldsToRead >= filterSize)
+                {
+                    const char* lockTypeStr = filterStringArray.item(i+1);
+                    if (lockTypeStr && *lockTypeStr && isdigit(*lockTypeStr))
+                        lockType = (DALockTypeFilter) atoi(lockTypeStr);
+                }
+                break;
+            case DALFTEndpoint:
+                if (filterFieldsToRead >= filterSize)
+                    endpoint.set(filterStringArray.item(i+1));
+                break;
+            case DALFTDuration:
+                filterSize = 3;
+                if (filterFieldsToRead >= filterSize)
+                {
+                    const char* durationLowStr = filterStringArray.item(i+1);
+                    if (durationLowStr && *durationLowStr && isdigit(*durationLowStr))
+                        durationLow = atoi(durationLowStr);
+                    const char* durationHighStr = filterStringArray.item(i+2);
+                    if (durationHighStr && *durationHighStr && isdigit(*durationHighStr))
+                        durationHigh = atoi(durationHighStr);
+                }
+                break;
+            }
+            filterFieldsToRead -= filterSize;
+            i += (filterSize - 1);
+        }
+    }
+    bool checkEndpoint(const char *_endpoint)
+    {
+        if (!_endpoint || !*_endpoint)
+            return false;
+        if (endpoint.length() && !streq(endpoint.get(), "*") && !WildMatch(_endpoint, endpoint.get(), true))
+            return false;
+        return true;
+    }
+    bool checkDuration(int duration)
+    {
+        if ((duration < durationLow) || ((durationHigh > -1) && (durationHigh < duration)))
+            return false;
+        return true;
+    }
+    bool checkAndReadPath(const char* xpath, bool& isFile, StringBuffer& file)
+    {
+        if (!xpath || !*xpath)
+            return false;
+
+        const char* ptr = xpath;
+        isFile = beginWith(ptr, "Files");
+        if ((isFile && (lockType == DALTFNonFileLock)) || (!isFile && (lockType == DALTFFileLock)))
+            return false;
+
+        file.clear();
+        //Build file name with path using scope names and file name
+        while (ptr && *ptr && (beginWith(ptr,"/Scope[@name=\"")
+            || beginWith(ptr,"/File[@name=\"") || beginWith(ptr,"/SuperFile[@name=\"")))
+        {
+            if (file.length())
+                file.append("::");
+            while (ptr && *ptr && (*ptr != '"'))
+                file.append(*(ptr++));
+            if (ptr && (*ptr == '"'))
+                ptr++;
+            if (ptr && (*ptr == ']')) //Skip ']' no matter whether it follows the '"' or not.
+                ptr++;
+        }
```

Comment:
A conversion from a logical filename's xpath form to it's standard form should be part of CDfsLogicalFileName really.
It can already go the other way..